### ISSUE TITLE
Performance issue with MiniZinc 2.7.0

### ISSUE THREAD

**Issue by scand1sk** [MiniZinc/libminizinc#645]
I use MiniZinc/OR-Tools to solve fairly large timetabling problems (attached).
With MiniZinc 2.6.4, the compilation phase took about 2 minutes on my computer.
With MiniZinc 2.7.0, the compilation phase is about 20 minutes.

As my model is too large to be solved to solutions of good enough quality in reasonable time, I try to improve solutions by solving fragments, à la LNS. This requires to launch the solver several times. Having a long compilation phase makes the approach unusable.

Is there an issue with my model, or is there some regression introduced in MiniZinc 2.7.0 ?

I launch MiniZinc from the command line with options "-p7 -f -a".

[timetable.zip](https://github.com/MiniZinc/libminizinc/files/10862661/timetable.zip)

**Comment by Dekker1** [comment]
We traced the difference back to f285a3b5dfd69d9368f58b3c8806fedea33f61f8, but still have to further investigate the problem to find the direct cause.

**Comment by guidotack** [comment]
I've pushed a fix to the develop branch now, which will be released as version 2.7.1 early next week.
You can speed up your model by working around a limitation in the `or-tools` MiniZinc library. It doesn't come with a propagator for `disjunctive`, so all your resource constraints need to be decomposed, which is quite expensive. The following line will rewrite the disjunctive into the native cumulative constraint in `or-tools`:

```
fzn_cumulative([start[s] | s in parRessource[g]], [durations[s] | s in parRessource[g]], [1 | s in parRessource[g]], 1)
```

This cuts down compilation time in half on my laptop (it doesn't seem to improve solving time too much though).
